compiler/modules/front.syntax/token.bal

```diff
@@ -499,6 +528,57 @@ public readonly class SourceFile {
     }
 }
 
+function tokenEndCodePointIndex(string[] fragments, FragCode[] fragCodes, int startCodePointIndex) returns int {
+    int fragmentIndex = fragmentCountUpTo(fragments, startCodePointIndex);
+    return lineTokenEndIndex(fragments, fragCodes, startCodePointIndex, fragmentIndex);
+}
+
+function fragmentCountUpTo(string[] fragments, int startCodePointIndex) returns int {
+    int nCodePoints = 0;
+    int fragmentIndex = 0;
+    int nFragments = fragments.length();
+    while fragmentIndex < nFragments {
+        if nCodePoints >= startCodePointIndex {
+            break;
+        }
+        nCodePoints += fragments[fragmentIndex].length();
+        fragmentIndex += 1;
+    }
+    return fragmentIndex;
+}
+
+// Return code-point index of end of token that starts in
```

Comment:
Comment is incomplete.

Reply:
function merged to `tokenEndCodePointIndex`